### ISSUE TITLE
Npm install failing in Github Actions

### ISSUE THREAD

**Issue by timwright12** [department-of-veterans-affairs/va-mobile-app#7188]
We're getting npm install failures in the iOS and Android builds recently that are forcing reruns of the workflows. 

The error:
> error An unexpected error occurred: "/home/runner/work/va-mobile-app/va-mobile-app/VAMobile/node_modules/core-js: Cannot read properties of undefined (reading 'config')".

We can get it to pass if we keep trying, but that's not a great solution

**Comment by timwright12** [comment]
Related ticket where adding retries is mentioned: https://app.zenhub.com/workspaces/va-mobile-60f1a34998bc75000f2a489f/issues/gh/department-of-veterans-affairs/va-mobile-app/7184

**Comment by theodur** [comment]
After looking into the failures, it seems to be caused by different reasons, because an error I've been seeing a lot lately is:

> info This package requires node-gyp, which is not currently installed. Yarn will attempt to automatically install it. If this fails, you can run "yarn global add node-gyp" to manually install it.
error An unexpected error occurred: "/home/runner/work/va-mobile-app/va-mobile-app/VAMobile/node_modules/core-js: Cannot read properties of undefined (reading 'getOption')".

Doing some research into the error above, it looks like npm removed the `node-gyp` binary which is causing this issue intermittently because yarn tries to install it. The breaking change has been reverted in npm, and should be available in the next npm release ([issue](https://github.com/npm/cli/pull/6554#issuecomment-1780237375)).

After looking into the other error:

> error An unexpected error occurred: "/home/runner/work/va-mobile-app/va-mobile-app/VAMobile/node_modules/core-js: Cannot read properties of undefined (reading 'config')".

It seems like this is also cause by the `node-gyp` binary being removed ([older similar issue](https://github.com/yarnpkg/yarn/issues/3905#issuecomment-315213840)).

Since the `node-gyp` binary is going to be added back soon, in the meantime I opened a [PR](https://github.com/department-of-veterans-affairs/va-mobile-app/pull/7218) adding retries for the yarn install steps. Once the binary is added back and released in npm, we may not need the retires anymore, but it can't hurt to keep them.
